plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/SocketAccess.java

```diff
@@ -44,33 +44,51 @@ private SocketAccess() {}
         try {
             return AccessController.doPrivileged(operation);
         } catch (PrivilegedActionException e) {
-            throw (IOException) e.getCause();
+            Throwable cause = e.getCause();
+            if (cause instanceof IOException) {
+                throw (IOException) e.getCause();
+            } else {
+                throw new IOException(cause);
+            }
         }
     }
 
-    public static <T> T doPrivilegedException(PrivilegedExceptionAction<T> operation) throws StorageException {
+    public static <T> T doPrivilegedException(PrivilegedExceptionAction<T> operation)
+            throws StorageException, IOException, URISyntaxException {
         SpecialPermission.check();
         try {
             return AccessController.doPrivileged(operation);
         } catch (PrivilegedActionException e) {
-            throw (StorageException) e.getCause();
+            safeRethrowCause(e);
```

Comment:
Maybe change `safeRethrowCause()` so that it returns an `Exception` and use throw `unwrapCause(e);` here?

Reply:
@tlrx the problem is that I can't really do that because we throw three different kinds of checked exceptions (then I'd have to adjust the signature to `throws Exception` and do even more noisy changes)?